src/Form/MediaLibraryFileUploadForm.php

```diff
@@ -0,0 +1,263 @@
+      return $response;
+    }
+    return parent::updateFormCallback($form, $form_state);
+  }
+
+  /**
+   * Submit handler for the upload button, below the dropzone area..
+   *
+   * @param array $form
+   *   The form render array.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   The form state.
+   */
+  public function uploadDropzoneSubmit(array $form, FormStateInterface $form_state) {
+    $media_type = $this->getMediaType($form_state);
+    $field_config = $media_type->getSource()
+      ->getSourceFieldDefinition($media_type);
+    $field_storage = $field_config->getFieldStorageDefinition();
+    $upload_destination = $field_storage->getSetting('uri_scheme') . '://' . $field_config->getSetting('file_directory');
+
+    $files = [];
+    foreach ($form_state->getValue(['dropzone', 'uploaded_files']) as $file) {
+      $files[] = $this->dropzoneSave->createFile($file['path'], $upload_destination, $form['container']['dropzone']['#extensions'], $this->currentUser);
+    }
+    $this->processInputValues($files, $form, $form_state);
+  }
+
+  /**
+   * {@inheritDoc}}
+   */
+  public function validateUploadElement(array $element, FormStateInterface $form_state) {
+    $form_state->setValue('upload', ['fids' => []]);
+    return parent::validateUploadElement($element, $form_state);
+  }
+
+  /**
+   * {@inheritDoc}}
+   */
+  protected function buildEntityFormElement(MediaInterface $media, array $form, FormStateInterface $form_state, $delta) {
+    $element = parent::buildEntityFormElement($media, $form, $form_state, $delta);
+
+    // If similar media items already exist, present those options to the user.
+    if ($similar_options = $this->getSimilarMediaOptions($media)) {
+      $element['similar_media']['#tree'] = TRUE;
+      // We don't set a default value on this field and we require it so that
+      // it forces the user to make a decision.
+      $element['similar_media'][$delta] = [
+        '#type' => 'radios',
+        '#required' => TRUE,
+        '#title' => $this->t('Possible similar items'),
+        '#description' => $this->t('These files already exist on the site. You can use one of these items or continue to upload a new file. These are only possible suggestions.'),
+        '#options' => $similar_options + [$this->t('Add new')],
+        '#weight' => -10,
+        '#attached' => ['library' => ['stanford_media/admin']],
+        '#prefix' => '<div class="similar-media-options">',
+        '#suffix' => '</div>',
+      ];
+    }
+    return $element;
+  }
+
+  /**
+   * Get the available media options for the radio buttons.
+   *
+   * @param \Drupal\media\MediaInterface $media
+   *   Media item to compare.
+   *
+   * @return array
+   *   Similar media rendered options for radios.
+   *
+   * @throws \Drupal\Component\Plugin\Exception\PluginException
+   */
+  protected function getSimilarMediaOptions(MediaInterface $media) {
```

Comment:
This service is still working wonderfully.

Reply:
So nice to build things that just continue to work